### DISCUSSION TITLE
Default Filters in Modal Windows

### DISCUSSION THREAD

**Discussion by Cattrah** [YetiForceCompany/YetiForceCRM#15975] [How to]
Is there a setting to apply a user's default filter to any related modal windows for selecting records?

For example. I create a custom products filter for a user and set it as their default filter
![image](https://user-images.githubusercontent.com/101741600/164323158-4697e887-ce23-462d-bfc3-6882c7736ed7.png)

Then when they access the products module from the menu, this filter is automatically selected for them. great! I love it!
![image](https://user-images.githubusercontent.com/101741600/164323316-8aeed0d0-0e80-4c66-a7a3-6b165de0076a.png)

But when they access a products list from a modal select window like when creating a new quote, the default filter is not selected, but it is in the dropdown list. Some of our users are not very technical and this would go a long way to making things easier for them.
![image](https://user-images.githubusercontent.com/101741600/164323544-25a1149b-3d7f-4f6b-9e82-0a2baf8ab4d6.png)

![image](https://user-images.githubusercontent.com/101741600/164323635-da610e73-660f-4173-b487-fffec572fbdf.png)

**Comment by rskrzypczak** [comment]
No, there are no such settings. By default, only fields are listed.

**Comment by vovpff** [comment]
U can add array of fields (field names) what you need to see into `\modules\Products\Products.php` into this `$search_fields_name`. There is some example:

```
	public $search_fields_name = [
		0 => 'engraving_name',
		1 => 'engraving_material',
		2 => 'engraving_thickness',
		3 => 'engraving_price',
		4 => 'engraving_price_ra',
		5 => 'engraving_price_vip',
		6 => 'engraving_price_cut',
		7 => 'engraving_price_cut_ra',
		8 => 'engraving_price_cut_vip',
		9 => 'engraving_price_engrave',
		10 => 'engraving_price_engrave_ra',
		11 => 'engraving_price_engrave_vip',
		12 => 'engraving_comment',
	];
```

**Reply by rskrzypczak** [reply]
Yes, but this applies only to the listed fields, it does not take into account the conditions set on the filter.
It is easier to modify the fields on the default filter itself than to interfere with the files.

**Comment by Cattrah** [comment]
Maybe I can modify the code on the modal to look at the logged in User's default filter for the module and preselect it in the list? It seems like all that information should be available at the time the modal is rendered. What file(s) is the modal rendered from?
